twine/utils.py

```diff
@@ -99,19 +101,44 @@ def get_config(path: str = "~/.pypirc") -> Dict[str, RepositoryConfig]:
     return dict(config)
 
 
+def validate_url(repository_url: Optional[str]) -> bool:
+    """Validate the given url for allowed schemes and components"""
+
+    if not repository_url:
+        return False
+
+    # Scheme should always be https, and the url should at minimum
+    # contain scheme and host
+    validator = (
+        validators.Validator()
+        .allow_schemes("https")
+        .require_presence_of("scheme", "host")
+    )
+
+    # Only return scheme when the url is valid
+    url = uri_reference(repository_url)
+    try:
+        validator.validate(url)
+    except Exception:
```

Comment:
Why are we handling `Exception` here? The code declares what exceptions it might raise: https://rfc3986.readthedocs.io/en/latest/api-ref/validators.html#rfc3986.validators.Validator.validate

Reply:
Please see https://github.com/pypa/twine/pull/602/files#r422648080